references/depth/stereo/cascade_evaluation.py

```diff
@@ -0,0 +1,242 @@
+
+        def preprocessing(image_left, image_right, disp, valid_disp_mask):
+            image_left, image_right = trans(image_left, image_right)
+            if disp is not None and not isinstance(disp, torch.Tensor):
+                disp = torch.from_numpy(disp)
+            if valid_disp_mask is not None and not isinstance(valid_disp_mask, torch.Tensor):
+                valid_disp_mask = torch.from_numpy(valid_disp_mask)
+            return image_left, image_right, disp, valid_disp_mask
+
+    else:
+        preprocessing = make_eval_transform(args)
+
+    metrics = {}
+    for name in val_datasets:
+        base_image_folder = os.path.join(image_checkpoint_folder, name)
+        os.makedirs(base_image_folder, exist_ok=True)
+        
+        metrics[name] = {}
+        val_dataset = make_dataset(name, args.dataset_root, transforms=preprocessing)
+        for n_cascades in args.num_cascades:
+            for n_iters in args.num_iters:
+                
+                config = f"{n_cascades}c_{n_iters}i"
+                config_image_folder = os.path.join(base_image_folder, config)
+                os.makedirs(config_image_folder, exist_ok=True)
+                
+                metrics[name][config] = _evaluate(
+                    model,
+                    args,
+                    val_dataset,
+                    iterations=args.recurrent_updates,
+                    padder_mode="kitti",
+                    header=f"{name} evaluation",
+                    batch_size=args.batch_size,
+                    writter=writter,
+                    step=step,
+                    iterations=n_iters,
+                    cascades=n_cascades,
+                )
+    
+    metric_log = []
+    # print the final results
+    for name in val_datasets:
+        for config in metrics[name]:
+            config_tokens = config.split("_")
+            config_iters = config_tokens[1][:-1]
+            config_cascades = config_tokens[0][:-1]
+            
+            evaluation_str = f"{name} evaluation@ size:{args.resize_size} n_cascades:{config_cascades} recurrent_updates:{config_iters}"
+            metrics_str = f"Metrics: {metrics[name][config]}"
+            metric_log.extend([evaluation_str, metrics_str])
+            
+            print(evaluation_str)
+            print(metrics_str)        
+
+    eval_log_name = f"{checkpoint_name.replace('.pth', '')}_eval.log"
+    print("Saving eval log to: ", eval_log_name)
+    with open(eval_log_name, "w") as f:
+        f.write("\n".join(metric_log))
+            
+
+def load_checkpoint(args):
+    checkpoint = torch.load(args.checkpoint, map_location=torch.device("cpu"))
+    experiment_args = checkpoint["args"]
+    model = torchvision.prototype.models.depth.stereo.__dict__[experiment_args.model](weights=None)
+    model.load_state_dict(checkpoint["model"])
+    return model
+
+    
+def main(args):
+    model = load_checkpoint(args)
+    evaluate(model, args, padder_mode="kitti")
```

Comment:
Do padder_mode always be `kitti`?

Reply:
If you think always using Kitti is fine, in this case I think we should remove `padder_mode` and always use the same padding mode without the need of input param.
Otherwise if we need different padder_mode depend on dataset, then I think we should add extra param in argparse.